1. soru

* çocuk sahibi olmayanlar olanlara gore  $ fazla eglenceye harcama yapmaktadır.
* yıllık gelir 1 birim arttıgında;

0.00004*yıllık gelir +0.00122 kadar fazla harcama olur

özelliklerin ve hedefin veri dağılımına ihtiyac vardır.



2. soru

In [10]:
import numpy as np
import pandas as pd
from sklearn import linear_model
import matplotlib.pyplot as plt
import plotly.graph_objs as go
import plotly.plotly as py
import scipy.stats as stats
import statsmodels.api as sm

import warnings
warnings.filterwarnings('ignore')

In [2]:
weather = pd.read_csv('weatherHistory.csv')
weather.head(10)

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   
5  2006-04-01 05:00:00.000 +0200  Partly Cloudy        rain         9.222222   
6  2006-04-01 06:00:00.000 +0200  Partly Cloudy        rain         7.733333   
7  2006-04-01 07:00:00.000 +0200  Partly Cloudy        rain         8.772222   
8  2006-04-01 08:00:00.000 +0200  Partly Cloudy        rain        10.822222   
9  2006-04-01 09:00:00.000 +0200  Partly Cloudy        rain        13.772222   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   
3                  5.944444      0.83            14.1036   
4                  6.977778      0.83            11.0446   
5                  7.111111      0.85            13.9587   
6                  5.522222      0.95            12.3648   
7                  6.527778      0.89            14.1519   
8                 10.822222      0.82            11.3183   
9                 13.772222      0.72            12.5258   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   
3                   269.0          15.8263         0.0               1016.41   
4                   259.0          15.8263         0.0               1016.51   
5                   258.0          14.9569         0.0               1016.66   
6                   259.0           9.9820         0.0               1016.72   
7                   260.0           9.9820         0.0               1016.84   
8                   259.0           9.9820         0.0               1017.37   
9                   279.0           9.9820         0.0               1017.22   

                       Daily Summary  
0  Partly cloudy throughout the day.  
1  Partly cloudy throughout the day.  
2  Partly cloudy throughout the day.  
3  Partly cloudy throughout the day.  
4  Partly cloudy throughout the day.  
5  Partly cloudy throughout the day.  
6  Partly cloudy throughout the day.  
7  Partly cloudy throughout the day.  
8  Partly cloudy throughout the day.  
9  Partly cloudy throughout the day.

In [5]:
weather['temp_diff']=weather['Temperature (C)']- weather['Apparent Temperature (C)']

In [4]:
weather.head()

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   
3  2006-04-01 03:00:00.000 +0200  Partly Cloudy        rain         8.288889   
4  2006-04-01 04:00:00.000 +0200  Mostly Cloudy        rain         8.755556   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   
3                  5.944444      0.83            14.1036   
4                  6.977778      0.83            11.0446   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   
3                   269.0          15.8263         0.0               1016.41   
4                   259.0          15.8263         0.0               1016.51   

                       Daily Summary  temp_diff  
0  Partly cloudy throughout the day.   2.083333  
1  Partly cloudy throughout the day.   2.127778  
2  Partly cloudy throughout the day.   0.000000  
3  Partly cloudy throughout the day.   2.344444  
4  Partly cloudy throughout the day.   1.777778

In [7]:
X=weather[['Humidity','Wind Speed (km/h)']]

In [8]:
Y=weather['temp_diff']

In [11]:
X = sm.add_constant(X)

In [12]:
sonuclar = sm.OLS(Y, X).fit()
sonuclar.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              temp_diff   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Sat, 25 May 2019   Prob (F-statistic):               0.00
Time:                        02:44:58   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -2.4381      0.021   -115.948      0.000      -2.479      -2.397
Humidity              3.0292      0.024    126.479      0.000       2.982       3.076
Wind Speed (km/h)     0.1193      0.001    176.164      0.000       0.118       0.121
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.264
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                           0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

istatistiksel olarak p ye baktıgımızda constant ve degerlerimizin  anlamlı oldugunu goruyoruz.
Temp diff= -2.4381+3.0292*Humidity+0.1193*Wind Speed (km/h)
* nem oranı bir arttıgında hissedilen sıcaklık 3.0292 artmaktadır.
* ruzgar hızı bir arttıgında hissedilen sıcaklık 0.1193 artmaktadır.
ruzgar hızı arttıkca sıcaklık hıssının azalmasını beklerdım, degerlerde eksık oldugundan katsayısı yanlıs cıkmıstır.

In [13]:
weather['Humidity_WindS']=weather['Humidity']*weather['Wind Speed (km/h)']

In [14]:
X=weather[['Humidity','Wind Speed (km/h)','Humidity_WindS']]

In [15]:
Y=weather['temp_diff']

In [16]:
X = sm.add_constant(X)

In [17]:
sonuclar = sm.OLS(Y, X).fit()
sonuclar.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              temp_diff   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Sat, 25 May 2019   Prob (F-statistic):               0.00
Time:                        02:54:59   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -0.0839      0.033     -2.511      0.012      -0.149      -0.018
Humidity             -0.1775      0.043     -4.133      0.000      -0.262      -0.093
Wind Speed (km/h)    -0.0905      0.002    -36.797      0.000      -0.095      -0.086
Humidity_WindS        0.2971      0.003     88.470      0.000       0.291       0.304
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.262
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                           0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

istatistiksel olarak p ye baktıgımızda constant ve degerlerimizin  anlamlı oldugunu goruyoruz.
Temp diff= -0.0839+-0.1775*Humidity+-0.0905*Wind Speed (km/h)+0.2971*Humidity_WindS
* nemdeki bir artıs, ruzgarda bir puan artanlarda artmayanlara göre 0.2971 artmaktadır
* ruzgar hızındaki bir artıs, nemde bir puan artanlarda artmayanlara göre  0.2971 artmaktadır.


işaretler değişti

3. soru